UI/signIn.tsx

```diff
@@ -60,174 +65,181 @@ export async function getCurrentSignInCtx() {
     return undefined;
 }
 
-export async function signInWithExtension() {
-    const albyCtx = await Nip7ExtensionContext.New();
-    if (albyCtx instanceof Error) {
-        return albyCtx;
-    }
-    if (albyCtx === undefined) {
-        open(AlbyURL);
-    }
-    return albyCtx;
-}
-
-export function signInWithPrivateKey(privateKey: PrivateKey) {
-    const ctx = InMemoryAccountContext.New(privateKey);
-    if (ctx instanceof Error) {
-        throw ctx;
-    }
-    localStorage.setItem("MPK", privateKey.hex);
-    setSignInState("local");
-    return ctx;
-}
-
 type Props = {
-    eventBus: EventEmitter<SignInEvent>;
-} & SignInModel;
-
-export type SignInModel = {
-    privateKey: string;
-    warningString?: string;
+    emit: emitFunc<SignInEvent>;
 };
 
-type State = { state: "newAccount" | "enterPrivateKey" };
+type State = {
+    state: "newAccount" | "enterPrivateKey";
+    privateKey: PrivateKey;
+    privateKeyError: string;
+    albyError: string;
+};
 export class SignIn extends Component<Props, State> {
-    render() {
-        const props = this.props;
+    styles = {
+        container:
+            tw`h-screen w-screen bg-[${PrimaryBackgroundColor}] flex items-center justify-center p-4 overflow-y-auto`,
+        form: tw`w-[30rem] flex flex-col h-full py-8`,
+        logo: tw`w-32 h-32 mx-auto`,
+        title: tw`text-[${PrimaryTextColor}] text-center text-4xl`,
+        subTitle: tw`text-[${HintTextColor}] text-center`,
+        input: tw`w-full px-4 py-2 focus-visible:outline-none rounded-lg mt-8`,
+        hint: tw`text-[${HintTextColor}] text-sm mt-2`,
+        block: tw`flex-1 desktop:hidden`,
+        signInButton:
+            tw`w-full mt-4 ${ButtonClass} ${LinearGradientsClass} hover:bg-gradient-to-l mobile:rounded-full font-bold`,
+        ablyButton:
+            tw`${ButtonClass} ${CenterClass} mt-4 bg-[#F8C455] text-[#313338] hover:bg-[#FFDF6F] py-0 mobile:hidden`,
+        cancelButton:
+            tw`${ButtonClass} ${CenterClass} mt-4 bg-[${SecondaryBackgroundColor}] text-[${PrimaryTextColor}] hover:bg-[${HoverButtonBackgroudColor}] mobile:rounded-full`,
+        ablyIcon: tw`h-10`,
+        isError: (error: string) => error ? tw`text-[${ErrorColor}]` : "",
+    };
+
+    signInWithPrivateKey = () => {
+        if (!this.state.privateKey && !this.state.privateKeyError) {
+            this.setState({
+                state: "newAccount",
+                privateKeyError: "",
+                privateKey: PrivateKey.Generate(),
+            });
+            return;
+        }
+
+        const ctx = InMemoryAccountContext.New(this.state.privateKey);
+        localStorage.setItem("MPK", this.state.privateKey.hex);
```

Comment:
Please write a local private key management module instead of inlining key handling logic.